src/Product/SearchProvider.php

```diff
@@ -83,24 +83,28 @@ public function __construct(
     }
 
     /**
+     * @param ProductSearchQuery $query
+     *
      * @return array
      */
-    private function getAvailableSortOrders()
+    private function getAvailableSortOrders($query)
```

Comment:
Not blocking, but if querytype is only what we need, we could ask for query type.

Reply:
I will pass the whole query, maybe we want to do other things with it?